ddtrace/internal/ci_visibility/encoder.py

```diff
@@ -79,3 +83,35 @@ def _convert_span(span, dd_origin):
         else:
             event_type = "span"
         return {"version": CIVisibilityEncoderV01.TEST_EVENT_VERSION, "type": event_type, "content": sp}
+
+
+class CIVisibilityCoverageEncoderV02(CIVisibilityEncoderV01):
+    PAYLOAD_FORMAT_VERSION = 2
+
+    def put(self, spans):
+        spans_with_coverage = [span for span in spans if COVERAGE_TAG_NAME in span.get_tags()]
```

Comment:
oh you're setting the span tag so that you can read the tags here? That's fine, but we should then remove the tag from the spans after we've encoded it

Reply:
from the event `citestcycle` intake.